python/tvm/relax/vm_build.py

```diff
@@ -310,6 +310,7 @@ def foo(x: Tensor((3, 4), "float32"), y: Tensor((3, 4), "float32")):
     passes.append(relax.transform.RemovePurityChecking())
     passes.append(relax.transform.CallTIRRewrite())
     passes.append(relax.transform.StaticPlanBlockMemory())
+    passes.append(relax.transform.LowerAllocTensor())
 
     if tvm.transform.PassContext.current().config.get("relax.backend.use_cuda_graph", False):
         passes.append(relax.transform.RewriteCUDAGraph())
```

Comment:
Did you look in to what assumptions `relax.transform.RewriteCUDAGraph` makes on static vs dynamic allocations to ensure the expectations match before and after this change, given that there is some dependence on alloc_tensor in RewriteCudaGraph?

Reply:
It appears to be a bug in `RewriteCUDAGraph`, which occurs when there is a `R.memory.alloc_storage` that is then used in a trivial var-to-var rebinding.

Reply:
This bug ended up being trickier to track down than I had expected.  It will be much simpler to solve after https://github.com/apache/tvm/pull/15810 lands, since the `.kill_*` methods won't be inserted yet.  For now, I've re-ordered the passes so the `LowerAllocTensor` occurs after `RewriteCUDAGraph`.

Reply:
Awesome, many thanks for looking into this @Lunderberg, changing the ordering until the fix is in makes sense. Let's land this an #15810.